### PR TITLE
neutral style in basics/storage-classes

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: basics/storage-classes.md [hunk 1]
```diff
@@ -19,19 +19,19 @@ allowed to change.
...
 
-`immutable` objects can thus safely be shared among different threads
+Thus `immutable` objects can safely be shared among different threads
 because they never change by design. This implies that `immutable`
 objects can be cached perfectly.
 
 ### `const`
 
-`const` objects can't be modified, too. This
+Objects of the storage class `const` can't be modified, too. This
```
Comment (ZombineDev) [inline basics/storage-classes.md]: tl;dr "const objects" sounds better to me and is technically more correct.

`const` is type qualifier / type constructor, not a storage class. The "storage class" term comes from C and is also semantically non-transitive. The possible storage classes are: `auto` (compiler should figure out whether to place the variable on the stack or in register) `register` (put in register, if possible) `static` (statically allocated) `extern` (externally allocated). Storage classes are often confused with type qualifiers / attributes in D, because syntactically are allowed in the same places (they are all [attributes](https://dlang.org/spec/attribute.html#Attribute)), but semantically only `static`, `extern` and `__gshared` are storage classses (`__gshared` because it acts like `static` for function local variables).

Comment (ZombineDev) [inline basics/storage-classes.md]: Which brings to the thought that this chapter should really be called "Mutability" or "Type qualifiers".

Comment (ZombineDev) [inline basics/storage-classes.md]: Also (real) storage classes is unnecessary technical terminology for the basics part of the tour. I think that they're better fit for my future memory chapter rewrite.

Comment (wilzbach) [inline basics/storage-classes.md]: > Which brings to the thought that this chapter should really be called "Mutability" or "Type qualifiers".

Please go for it! I just skimmed over the entire tour and searched for typos.

File: basics/storage-classes.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ to ensure they don't modify the input.
     const int* pb = &b;
     *pa = 7; // disallowed
 
-Both `immutable` and `const` are _transitive_ which ensures that once
+Both `immutable` and `const` are _transitive_ storage classes, which ensures that once
```
Comment (ZombineDev) [inline basics/storage-classes.md]: please change to "attributes" or "type qualifiers"

File: basics/storage-classes.md [hunk 3]
```diff
@@ -19,19 +19,19 @@ allowed to change.
...
 restriction is just valid for the current scope. A `const`
-pointer can be created from either a *mutable* or
+pointer can be created from either a *mutable* or an
 `immutable` object. This means that the object
-is `const` for your current scope, but someone
-else might modify it in future. Just with an `immutable`
-you will be sure that an object's value will never
+is `const` for the current scope, but someone
+else might modify it in future. Only with an `immutable`
+it is guaranteed that an object's value will never
```
Comment (ZombineDev) [inline basics/storage-classes.md]: "Only `immutable` objects are guaranteed to never change their value (after construction)."

File: basics/storage-classes.md [hunk 4]
```diff
@@ -19,19 +19,19 @@ allowed to change.
...
-`const` objects can't be modified, too. This
+Objects of the storage class `const` can't be modified, too. This
 restriction is just valid for the current scope. A `const`
-pointer can be created from either a *mutable* or
+pointer can be created from either a *mutable* or an
 `immutable` object. This means that the object
-is `const` for your current scope, but someone
-else might modify it in future. Just with an `immutable`
-you will be sure that an object's value will never
+is `const` for the current scope, but someone
```
Comment (ZombineDev) [inline basics/storage-classes.md]: Maybe it would be better to use the term "const reference" to an object, because const object are actually immutable.

**Comment by wilzbach** [comment]
>tl;dr "const objects" sounds better to me and is technically more correct.
> `const` is type qualifier / type constructor, not a storage class. The "storage class" term comes from C and is also semantically non-transitive. The possible storage classes are: `auto` (compiler should figure out whether to place the variable on the stack or in register) `register` (put in register, if possible) `static` (statically allocated) `extern` (externally allocated). Storage classes are often confused with type qualifiers / attributes in D, because syntactically are allowed in the same places (they are all [attributes](https://dlang.org/spec/attribute.html#Attribute)), but semantically only `static`, `extern` and `__gshared` are storage classses (`__gshared` because it acts like `static` for function local variables).

Replaced storage class with type qualifier.

> Which brings to the thought that this chapter should really be called "Mutability" or "Type qualifiers".

Done.

> Also (real) storage classes is unnecessary technical terminology for the basics part of the tour. I think that they're better fit for my future memory chapter rewrite.

Hehe - still looking forward ;-)
